Fix CNAME creation in win_dns_record

JLE-ATS opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
As suggested in https://github.com/ansible-collections/community.windows/issues/429, I have added a case to handle CNAME creation when the cname is part of a subdomain.
This is my first contribution and I am not used to the contribution process, so I am sorry if this is not the right way to submit fixes.
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
Fixes #429

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
win_dns_record

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
If this fix is accepted, the documentation should be updated with instructions for creating aliases in subdomains.
Unfortunately, I didn't find where I could update the documentation in this repository.
The following task could be included in the examples on the documentation page :
<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below
- name: Create the alias helloworld.test.example.com pointing to myserver.example.com
    community.windows.win_dns_record:
      name: "helloworld.test"
      type: "CNAME"
      value: "myserver.example.com"
      zone: "example.com"
```

jborean93 commented:
Just closing and re-opening to clear out the CI problems.

jborean93 commented on plugins/modules/win_dns_record.ps1:
```diff
@@ -150,11 +150,14 @@ if ($null -ne $values -and $values.Count -gt 0) {
                 Add-DnsServerResourceRecord -TXT -Name $name -DescriptiveText $value -ZoneName $zone -TimeToLive $ttl @extra_args -WhatIf:$module.CheckMode
             }
+            elseif ($type -eq 'CNAME') {
+                Add-DnsServerResourceRecordCName -Name $name -HostNameAlias $value -ZoneName $zone -TimeToLive $ttl -WhatIf:$module.CheckMode
```
So the changes here seem ok I'm just not sure what to do about the extra arguments like `@extra_args`, `-AllowUpdateAny` and possibly `@splat_args` that are in use in the original call.

`extra_args` contains the `-ComputerName` parameter which `Add-DnsServerResourceRecordCName` does have so it should be safe to add that. I think `splat_args` can probably be ignored as  it contains the CName info that this new cmdlet implicitly knows. I have no idea what `-AllowUpdateAny` update any does but the fact that `Add-DnsServerResourceRecord` also specified it makes me think this new call should.

JLE-ATS replied:
Thanks for your feedback.

- I have added the "@extra_args" to the command. I have also added a comment near the variable declaration to prevent ulterior breaking changes.

- I have not added the "-AllowUpdateAny" for now. This parameter has the same behaviour for both powershell modules so we could add it, but it allows any authenticated users to modify the DNS record which seems as a security problem. Maybe we could add a parameter to the ansible module to allow enabling/disabling this flag. What's your thoughts ?

- I have not added the "@splat_args" based on your comment and the fact that I did not find what this argument does. Is it doing anything at all ? 

- I have added a fragment as requested.

jborean93 reviewed (changes requested):
Are you able to add a changelog fragment as per https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment.

jborean93 commented:
Ignore the sanity failure, it will be fixed with https://github.com/ansible-collections/community.windows/pull/454 and is unrelated to your changes here.

JLE-ATS commented:
Thanks for your feedback.

- I have added the "@extra_args" to the command. I have also added a comment near the variable declaration to prevent ulterior breaking changes.

- I have not added the "-AllowUpdateAny" for now. This parameter has the same behaviour for both powershell modules so we could add it, but it allows any authenticated users to modify the DNS record which seems as a security problem. Maybe we could add a parameter to the ansible module to allow enabling/disabling this flag. **What's your thoughts ?**

- I have not added the "@splat_args" based on your comment and the fact that I did not find what this argument does. Is it doing anything at all ? 

- I have added a fragment as requested.
